Changes

```diff
@@ -251,6 +251,13 @@ Working version
   information before the first load.
   (Daniel Bünzli, review by Xavier Leroy and Nicolás Ojeda Bär)
 
+* #9757, #9846: check proper ownership when operating over mutexes.
+  Now, unlocking a mutex held by another thread or not locked at all
+  reliably raises a Sys_error exception.  Before, it was undefined
+  behavior, but the documentation did not say so.
```

Comment:
Note that the original report was about locking a mutex twice from the same thread (e.g. from an async callback). I propose to add the various informations from my own patch:
```
  With systhread, mutexes now raise an exception when
  attempting to lock recursively, instead of deadlocking. This is
  meant to improve the failure mode when locking a mutex or a channel
  in a finaliser or a signal handler, that are already locked by the
  thread. On Windows, mutexes used to be accidentally recursive, so
  this might break Windows-specific programs that relied on this
  behaviour.
```

Reply:
Since this is a breaking change, I wanted the Changes entry to emphasize what is broken most, namely, using Mutex.unlock from another thread.

Reply:
Thanks, the new description looks fine.